firebase/functions/index.js

```diff
@@ -55,12 +121,20 @@ function handleTestResult(origin, siteName, message) {
   const siteDoc = db.collection('site').doc(siteName);
   const portDoc = originDoc.collection('port').doc('port' + message.port);
   const deviceDoc = originDoc.collection('device').doc(message.device_id);
-  return Promise.all([
+  const updates = [
     originDoc.set({ 'updated': timestamp }),
     siteDoc.set({ 'updated': timestamp }),
     portDoc.set({ 'updated': timestamp }),
     deviceDoc.set({ 'updated': timestamp })
-  ]).then(() => {
+  ];
+  if (message.config && message.config.run_info
+    && message.config.run_info.expiration) {
+    const expirationDoc = originDoc.collection('expiration').doc(message.runid);
```

Comment:
I think the DB schema has gotten more complicated with some of these changes, and it's getting to the point where I think we should document it since there's getting to be more and more dependencies between various parts of the setup.  I don't inherently understand why this expiration time can't be a property of the data itself -- I'm sure there's a good reason for it, but something we'll have to make more explicit, I think.